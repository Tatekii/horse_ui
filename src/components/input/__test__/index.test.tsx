import { mountTest } from "@/test/shared";
import { render, screen } from "@testing-library/react";
import Input from "@/components/input";
import Icon from "@/components/icon";

describe("Testing Input", () => {
  mountTest(() => <Input />);

  it("should render default input", () => {
    render(<Input />);
    const target = screen.getByTestId("input-wrapper");
    expect(target).toMatchSnapshot();
  });

  it("should support size", () => {
    render(<Input size="large" />);
    const target = screen.getByTestId("input-wrapper");
    expect(target).toHaveClass("input-size-lg");
    expect(target).toMatchSnapshot();
  });

  it("should support icon", () => {
    render(<Input icon="coffee" />);
    const target = screen.getByTestId("icon");
    expect(target).toBeInTheDocument();
  });

  it("should support prepend and append", () => {
    render(
      <>
        <Input prepend="prepend" />
        <Input append={<Icon icon={"search"} />}></Input>
      </>
    );
    const icon = screen.getByTestId('icon')
    const target = screen.getByText('prepend')
    expect(icon).toBeInTheDocument()
    expect(target).toBeInTheDocument()
  });
});
