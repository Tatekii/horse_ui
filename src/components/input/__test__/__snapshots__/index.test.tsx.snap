// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input should render default input 1`] = `
<div
  class="input-wrapper"
  data-testid="input-wrapper"
>
  <input
    class="input-inner"
  />
</div>
`;

exports[`Testing Input should support size 1`] = `
<div
  class="input-wrapper input-size-lg"
  data-testid="input-wrapper"
>
  <input
    class="input-inner"
  />
</div>
`;
